### PR TITLE
feat: sped up lookup for revocation registries

### PR DESCRIPTION
**Pull request by wadeking98**
Previously when fetching multiple revocation registries, afj would wait for the response from the first request before sending off the second request. I added functionality to send off all requests at once and then wait for all responses. This makes the code more async and faster

### CODE REVIEW

File: packages/anoncreds/src/utils/getRevocationRegistries.ts [hunk 1]
```diff
@@ -76,28 +77,30 @@ export async function getRevocationRegistriesForRequest(
...
+              const { tailsFilePath } = await downloadTailsFile(agentContext, tailsLocation, tailsHash)
+
+              // const tails = await this.indyUtilitiesService.downloadTails(tailsHash, tailsLocation)
+              revocationRegistries[revocationRegistryId] = {
+                definition: revocationRegistryDefinition,
+                tailsFilePath,
+                revocationStatusLists: {},
+              }
+            }
+          })()
```
Comment (TimoGlastra) [inline packages/anoncreds/src/utils/getRevocationRegistries.ts:102]:
```suggestion
	const getRevocationRegistry = async () => {
            // Fetch revocation registry definition if not in revocation registries list yet
            if (!revocationRegistries[revocationRegistryId]) {
              const { revocationRegistryDefinition, resolutionMetadata } =
                await registry.getRevocationRegistryDefinition(agentContext, revocationRegistryId)
              if (!revocationRegistryDefinition) {
                throw new AriesFrameworkError(
                  `Could not retrieve revocation registry definition for revocation registry ${revocationRegistryId}: ${resolutionMetadata.message}`
                )
              }

              const { tailsLocation, tailsHash } = revocationRegistryDefinition.value
              const { tailsFilePath } = await downloadTailsFile(agentContext, tailsLocation, tailsHash)

              // const tails = await this.indyUtilitiesService.downloadTails(tailsHash, tailsLocation)
              revocationRegistries[revocationRegistryId] = {
                definition: revocationRegistryDefinition,
                tailsFilePath,
                revocationStatusLists: {},
              }
            }
          })
        revocationRegistryPromises.push(getRevocationRegistry())
```

File: packages/anoncreds/src/utils/getRevocationRegistries.ts [hunk 2]
```diff
@@ -164,38 +169,42 @@ export async function getRevocationRegistriesForProof(agentContext: AgentContext
...
-      const { revocationRegistryDefinition, resolutionMetadata } = await registry.getRevocationRegistryDefinition(
-        agentContext,
-        revocationRegistryId
-      )
-      if (!revocationRegistryDefinition) {
-        throw new AriesFrameworkError(
-          `Could not retrieve revocation registry definition for revocation registry ${revocationRegistryId}: ${resolutionMetadata.message}`
-        )
-      }
+    revocationRegistryPromises.push(
```
Comment (TimoGlastra) [inline packages/anoncreds/src/utils/getRevocationRegistries.ts:172]: same here, I find immediately executing function a bit hard to read often. cAn we extract to a separate method?

**Review by TimoGlastra — COMMENTED** [review]
Nice improvement! Some small code formatting suggestion

File: packages/anoncreds/src/utils/getRevocationRegistries.ts (hunk 1 above)
Comment (wadeking98) [reply packages/anoncreds/src/utils/getRevocationRegistries.ts]: got it, change completed

**Review by TimoGlastra — APPROVED** [review]

**Pull request merged**